core/storage/topic/gae_models.py

```diff
@@ -424,19 +425,9 @@ def has_reference_to_user_id(cls, user_id):
         Returns:
             bool. Whether any models refer to the given user ID.
         """
-        more_results = True
-        cursor = None
-        while more_results:
-            snapshot_content_models, cursor, more_results = (
-                cls.SNAPSHOT_CONTENT_CLASS.query().fetch_page(
-                    base_models.FETCH_BATCH_SIZE, start_cursor=cursor))
-            for snapshot_content_model in snapshot_content_models:
-                reconstituted_model = cls(**snapshot_content_model.content)
-                if user_id in reconstituted_model.manager_ids:
-                    return True
-        return (cls.query(cls.manager_ids == user_id).get(
-            keys_only=True) is not None or
-                cls.SNAPSHOT_METADATA_CLASS.exists_for_user_id(user_id))
+        return (cls.query(
+            cls.manager_ids == user_id).get(keys_only=True) is not None
+                or cls.SNAPSHOT_METADATA_CLASS.exists_for_user_id(user_id))
```

Comment:
These changes were already done in the exp and col models before, that is why the changes differ a little bit for the topic model.